Sources/Boundaries/Store.swift

```diff
@@ -1,9 +1,13 @@
 import Dispatch
+import Foundation
 import Optics
 import Prelude
 
-public final class Store<S, A> {
-  let reducer: Reducer<S, A>
+public final class Store<S, E: EffectProtocol> {
+  public typealias A = E.A
```

Comment:
now since the store is the global view of the application, this is where we require that the effects produce actions that the store understands.